### PR TITLE
Updates for 128x128, optimizer config

### PR DESCRIPTION
**Pull request by sgreenbury**
This pull request introduces several new experiment configuration files for different autoencoder and ensemble probabilistic diffusion (EPD) models, expands and updates datamodule configurations, and makes improvements to optimizer and dataloader settings. The changes enhance experiment reproducibility, flexibility in model training, and compatibility with various distributed training setups.

**Experiment Configuration Additions and Enhancements:**

* Added new experiment YAML files for autoencoder and EPD models, including variants with periodic boundary conditions, different normalization settings, and various loss functions and noise configurations. These provide ready-to-use templates for running experiments under different scenarios. [[1]](diffhunk://#diff-234b398f5a9e118e2767c9398744391b885d6f3a5ff5e93fbb144c6b99a878fbR1-R26) [[2]](diffhunk://#diff-b6dc8e41ce612389faad4252de155d0899c6977d8acf2feefcd559e1d2e6b909R1-R31) [[3]](diffhunk://#diff-ed13d09067e94947673e55d5fb0153d6fcb54334ddde5319d693ea4f3a071e39R1-R33) [[4]](diffhunk://#diff-3a411f1d2b1bcb48edc60ec36c69814cec288d113d389c46941b31cde91e9d6dR1-R34) [[5]](diffhunk://#diff-0b8e8b167690f9cc3a3a8bdec739a74c8ce3df36c85c638e40ded70bc30ffc6bR1-R34) [[6]](diffhunk://#diff-845ccae1ac856ef7cb0b36b37a5a7c6a02586f636aac9882261c05b2f813d8adR1-R34) [[7]](diffhunk://#diff-71628dd161240088c024a6ed09750185884865ae2ea6993af6cf88e473146727R1-R34) [[8]](diffhunk://#diff-7131e7fc97fe5c18463e85d0b15294782c590653e6d1e63ea69c744777f8d9c9R1-R36)

**Datamodule and Dataset Configuration Updates:**

* Introduced new datamodule configs for `conditioned_navier_stokes_128`, `gpe_laser_only_wake_128`, and `shallow_water2d_128`, and updated the data path for `gpe_laser_only_wake` to use a new dataset version. These changes support experiments on new or updated datasets with consistent normalization and loader settings. [[1]](diffhunk://#diff-97f9984477fd1c8f54f56ea7c0bf2c9f076bd589bd5d3003f8b9a664316866aeR1-R10) [[2]](diffhunk://#diff-ea4466adab2382fa3a13ea9efcd9b61b0c835fb6a49b927f8bc4c9971ed54c3cR1-R10) [[3]](diffhunk://#diff-cd1d6d6a1fec720f3892177d5d041eb306dc836f04890beccbbbea2d5fbf8873R1-R10) [[4]](diffhunk://#diff-e49ab6200e7493f82b9476cd28cde4292ab90ea14aa937d6e00383a4b0ca2214L2-R2)

**Dataloader Performance Improvements:**

* Enabled `pin_memory=True` in all DataLoader instantiations within `SpatioTemporalDataModule`, which can improve data transfer performance when using GPUs. [[1]](diffhunk://#diff-a8d7ed1915d2fcc36d2128b6c1bc77d97532a6e745f9fbaa624fe8446bc45e2cR313) [[2]](diffhunk://#diff-a8d7ed1915d2fcc36d2128b6c1bc77d97532a6e745f9fbaa624fe8446bc45e2cR324) [[3]](diffhunk://#diff-a8d7ed1915d2fcc36d2128b6c1bc77d97532a6e745f9fbaa624fe8446bc45e2cR341) [[4]](diffhunk://#diff-a8d7ed1915d2fcc36d2128b6c1bc77d97532a6e745f9fbaa624fe8446bc45e2cR352) [[5]](diffhunk://#diff-a8d7ed1915d2fcc36d2128b6c1bc77d97532a6e745f9fbaa624fe8446bc45e2cR369)

**Optimizer and Training Infrastructure Improvements:**

* Added a new optimizer config `adamw_half.yaml` for compatibility with external projects, and provided detailed comments for scheduler control in `adamw.yaml`, allowing more flexibility in learning rate scheduling. [[1]](diffhunk://#diff-e5f7c749e7a6fd0a415c32ba4c6d642e9e3262a7f7f7bb53f5377379dc87593eR1-R24) [[2]](diffhunk://#diff-eb7c74193e408d76f78eae5f70ed624682aac88f3999df0a926b8c8c9772733dR8-R18)
* Updated the single GPU SLURM distributed config to explicitly set `gpus_per_node`, `tasks_per_node`, and `ntasks` for improved SLURM job control.

**Loss Function Module Update:**

* Updated `autocast.losses.__init__.py` to include `EnsembleMAELoss` in the import and `__all__` list, making it available for use in experiment configs.

**Other Minor Improvements:**

* Updated `pyproject.toml` to specify test paths and directories to ignore for pytest, improving test discovery and isolation.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -123,9 +123,20 @@ torchvision = [
...
+norecursedirs = [
+    "outputs",
+    "slurm_scripts",
+]
 filterwarnings = [
     # Ignore Lightning warnings that are expected/benign in test environment
     "ignore:You are trying to `self.log\\(\\)` but the `self.trainer` reference is not registered:UserWarning",
     "ignore:GPU available but not used:UserWarning",
     "ignore:The '.*_dataloader' does not have many workers:UserWarning",
+    "ig"
```
Comment (sgreenbury) [inline pyproject.toml]:
```suggestion
```
